cnf-certification-test/networking/suite.go

```diff
@@ -67,8 +80,103 @@ var _ = ginkgo.Describe(common.NetworkingTestKey, func() {
 	ginkgo.It(testID, ginkgo.Label(testID), func() {
 		testMultusNetworkConnectivity(&env, defaultNumPings, netcommons.IPv6)
 	})
+	// Default interface ICMP IPv6 test case
+	testID = identifiers.XformToGinkgoItIdentifier(identifiers.TestUndeclaredContainerPortsUsage)
+	ginkgo.It(testID, ginkgo.Label(testID), func() {
+		testListenAndDeclared(&env)
+	})
 })
 
+func parseListening(res string, listeningPorts map[key]string) {
+	var k key
+	lines := strings.Split(res, "\n")
+	for _, line := range lines {
+		fields := strings.Fields(line)
+		if !strings.Contains(line, "LISTEN") {
+			continue
+		}
+		if indexprotocolname > len(fields) || indexport > len(fields) {
+			return
+		}
+		s := strings.Split(fields[indexport], ":")
+		p, _ := strconv.Atoi(s[1])
+		k.port = p
+		k.protocol = strings.ToUpper(fields[indexprotocolname])
+		k.protocol = strings.ReplaceAll(k.protocol, "\"", "")
+		listeningPorts[k] = ""
+	}
+}
+
+func checkIfListenIsDeclared(listeningPorts, declaredPorts map[key]string) map[key]string {
+	res := make(map[key]string)
+	if len(listeningPorts) == 0 {
+		return res
+	}
+	for k := range listeningPorts {
+		_, ok := declaredPorts[k]
+		if !ok {
+			tnf.ClaimFilePrintf(fmt.Sprintf("The port %d on protocol %s in pod %s is not declared.", k.port, k.protocol, listeningPorts[k]))
+			res[k] = listeningPorts[k]
+		}
+	}
+	return res
+}
+
+//nolint:funlen
+func testListenAndDeclared(env *provider.TestEnvironment) {
+	var k key
+	declaredPorts := make(map[key]string)
+	listeningPorts := make(map[key]string)
```

Comment:
Both declaredPorts and listeningPorts variables are initialized here but they're never cleared, so once a port is added to any of this map, it will be used for the rest of the containers in the loop. You should remove them from here and declare them at the begining of the containers loop in order to use empty maps on each iteration.
```
...
for _, cut := range env.Containers {
    declaredPorts := make(map[key]string)
	listeningPorts := make(map[key]string)
    ...
```

Reply:
I agree.